src/JsonSchema/Validator.php

```diff
@@ -52,4 +52,17 @@ public function coerce(&$value, $schema = null, JsonPointer $path = null, $i = n
 
         $this->addErrors(array_unique($validator->getErrors(), SORT_REGULAR));
     }
+
+    /**
+     * Does everything that coerce does, but will also set values to their default, if the value is not
+     * set and a default is available in the schema. Note that the first argumen is passwd by
+     * reference, so you must pass in a variable.
+     *
+     * {@inheritDoc}
+     */
+    public function coerceDefault(&$value, $schema = null, JsonPointer $path = null, $i = null)
```

Comment:
Yeah, that's a good point. Tell you what--as long as we're going to change the API, let's go all the way and just get rid of `coerce`, and convert that (back) to a config flag, as well. Then we'll modify `check` to just take a reference everywhere, and we'll be all set. I didn't do that on my second coercion attempt because I didn't want to break compatibility, in other words, this kind of thing would break:

```PHP
$validator->check(['foo'=>'bar'], $schema);
```
But now that I think about it, that's a pretty outlandish use case; I mean, I literally can't think of any rational reason to be doing it, outside of writing sample code for readmes and whatnot.

What do you think? That will should simplify things for us both. If you're game you can make the change in this PR, or I can do it when I move `$checkmode` back up into `Constraint` (which you're also welcome to do if you're feeling ambitious).